oscarapi/views/basket.py

```diff
@@ -92,13 +98,20 @@ def post(self, request, format=None):
             quantity = p_ser.validated_data['quantity']
             options = p_ser.validated_data.get('options', [])
 
-            basket_valid, message = self.validate(basket, product, quantity, options)
+            basket_valid, message = self.validate(
+                basket, product, quantity, options)
             if not basket_valid:
                 return Response(
                     {'reason': message},
                     status=status.HTTP_406_NOT_ACCEPTABLE)
 
-            basket.add_product(product, quantity=quantity, options=options)
+            line, created = basket.add_product(
+                product, quantity=quantity, options=options)
+
+            delete_line = overridable('OSCARAPI_DELETE_EMPTY_LINES', False)
+            if delete_line and line.quantity == 0:
+                line.delete()
```

Comment:
Well, I could adjust this, but it's not possible to have a negative quantity anymore:

        # don't allow quantities to be negative
        if basket.product_quantity(product) + quantity < 0:
            return False, _("Negative quantities are not allwed")
        return True, None